Cloud Pub/Sub Notifications support

akm opened the pull request:
From #47 

This PR have `blocks-gcs-proxy` support [Cloud Pub/Sub Notifications for Google Cloud Storage](https://cloud.google.com/storage/docs/pubsub-notifications).
Add [an example](https://github.com/groovenauts/blocks-gcs-proxy/blob/features/cloud_pubsub_notification_support/examples/pubsub_notifications/README.md) to try this feature.

See the [document](https://github.com/groovenauts/blocks-gcs-proxy/blob/features/cloud_pubsub_notification_support/doc/pubsub_notification.md) for more detail.

akm committed:
:+1: Bump up the version for Cloud Pub/Sub Notification support

akm committed:
:green_heart: Add test for Cloud Pub/Sub Notifications for Google Cloud Storage

akm committed:
:+1: Implement support of Cloud Pub/Sub Notifications for Google Cloud Storage

akm committed:
:pencil2: Remove descriptions about bundler

akm committed:
:shower: Remove items for ruby environment from .gitignore

akm committed:
:pencil2: Add document named "Use blocks-gcs-proxy with Cloud Pub/Sub Notifications"

akm committed:
:+1: Add an example with Pub/Sub Notifications

akm committed:
:+1: Add darwin to build targets for debug

akm committed:
:pencil2: Create notification with `-e OBJECT_FINALIZE`

akm committed:
:bug: Fix the return object type of JobMessage.DownloadFiles

akm committed:
:tada: Bump up the version

akm committed:
:pencil2: Add links to the document about pub/sub notifications to README.md

akm committed:
:pencil2: Fix unclosed block

akm committed:
:shirt: go fmt

minimum2scp commented on examples/pubsub_notifications/README.md:
````diff
@@ -0,0 +1,50 @@
+```
+
+gsutil notification create -t projects/scenic-doodad-617/topics/akm-pipeline01-job-topic                -f json gs://blocks-gcs-proxy-test -e OBJECT_FINALIZE
+gsutil notification create -t projects/scenic-doodad-617/topics/akm-test-gcs-pubsub-notifications-topic -f json gs://akm-test
````
Are last 2 lines necessary?

akm replied:
@minimum2scp  oops! I removed them. Thank you!

guemon commented on examples/pubsub_notifications/Dockerfile:
```diff
@@ -0,0 +1,19 @@
+COPY . $APP_HOME
+WORKDIR $APP_HOME
+
+ENV BLOCKS_GCS_PROXY_VERSION 0.6.0-alpha1
```
`0.6.0-alpha1` -> `0.6.0`

akm replied:
@guemon Thanks! I fixed it.

akm committed:
:pencil2: Remove working memo

akm committed:
:up: Use 0.6.0 instead of 0.6.0-alpha1 which includes a bug

ko1nksm commented on Makefile:
```diff
@@ -28,7 +28,9 @@ endif
-	gox -output="${PKGDIR}/{{.Dir}}_${OS}_${ARCH}" -os="${OS}" -arch="${ARCH}"
+	for x in "$(OS_LIST)" ; do \
+		gox -output="${PKGDIR}/{{.Dir}}_{{.OS}}_{{.Arch}}" -os="$$x" -arch="${ARCH}" ; \
+	done
```
@akm 

```for x in "$(OS_LIST)" ; do \``` expand to  ```for x in "linux darwin" ; do \```
therefore always loops one time.

gox's -os parameter accepts multiple values saparated by space.

> gox --help
>
> -os=""              Space-separated list of operating systems to build for

so, this code will work well, but meaningless "for".

akm replied:
@ko1nksm Thanks, I removed `for` statement.

akm committed:
:recycle: Pass $OS_LIST to -os option directly instead of using for statement

Because gox supports Space-separated list

$ gox --help
(snip)
  -os=""              Space-separated list of operating systems to build for

akm committed:
:shower: Remove unnecessary semicolon and backslash
